test-alerts.cpp

```diff
@@ -4,5 +4,39 @@
 #include "typewise-alert.h"
 
 TEST_CASE("infers the breach according to limits") {
-  REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+  REQUIRE(inferBreach(-1, 0, 30) == TOO_LOW);
+  REQUIRE(inferBreach(-1 , 0, 45) == TOO_LOW);
+  REQUIRE(inferBreach(-1 , 0, 40) == TOO_LOW);
+  REQUIRE(inferBreach( 0 , 0, 30) == NORMAL);  
+  REQUIRE(inferBreach( 31, 0, 30) == TOO_HIGH);
+  REQUIRE(inferBreach( 46, 0, 45) == TOO_HIGH);
+  REQUIRE(inferBreach( 41, 0, 40) == TOO_HIGH);
+  REQUIRE(inferBreach( 29, 0, 30) == NORMAL);  
+}
+
+TEST_CASE("Classify the temperature breach based on cooling type")
+{
+  REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING,-1) == TOO_LOW);
+  REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING,0) == NORMAL);
+  REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING,31) == TOO_HIGH);
+  REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING,-1) == TOO_LOW);
+  REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING,0) == NORMAL);
+  REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING,46) == TOO_HIGH);
+  REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING,-1) == TOO_LOW);
+  REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING,0) == NORMAL);
+  REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING,41) == TOO_HIGH);
+}
+
+TEST_CASE("Alerter")
+{
+  BatteryCharacter batteryChar;
+  batteryChar.coolingType=PASSIVE_COOLING;
+  REQUIRE(checkAndAlert(TO_EMAIL,batteryChar,50,"abc@xyz.com") == true);
```

Comment:
The test case is not effective, because  is not possible to check or verify if the alert was sent to email or to controller. Since return value is true, it is a easy overlook. this can be refactored further to add a stub for controller or email client, and the data received by the stub can be verified.

Same applies to line 39,40 and 41